python/mobitex_to_datablocks.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2025 Fabian P. Schmidt <kerel@mailbox.org>
+#
+# This file is part of gr-satellites
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+
+import itertools
+
+from gnuradio import gr
+
+import pmt
+import numpy as np
+
+from .check_eseo_crc import crc16_ccitt_zero as crc16_ccitt_zero
+from .mobitex_fec import decode, encode, Status
+
+
+def decode_control(control0: int, control1: int, fec: int) -> \
+        tuple[list[int], int] | None:
```

Comment:
This can be written as `Optional[tuple[list[int], int]]`. I've checked the [docs](https://docs.python.org/3/library/typing.html#typing.Optional) and apparently `Optional` is older. The `X | None` syntax appeared in Python 3.10. So here `Optional` would be preferred to support older Python versions, although I don't know what happens with type annotations that are unsupported by an older Python. Maybe they are ignored.

This `X | None` -> `Optional[X]` can be applied to a couple other places in the code.

Reply:
oh yes, if we target Python <=3.10, then the union type syntax is not supported yet.

 According to [PEP604](https://peps.python.org/pep-0604/):
> Optional values should be equivalent to the new union syntax

I've replaced all occurrences of the union type with `typing.Optional` now.

We could argue if we want type annotations at all, when it is not possible to use them with the latest (and hopefully stabilized) syntax. `typing.Optional` is a standard Python feature though, so I think it is OK to use (even though this would be the first occurrence of it in gr-satellites).